src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -722,22 +724,26 @@ impl<'a, 'tcx> Visitor<'tcx> for Qualifier<'a, 'tcx, 'tcx> {
                     // it means that our "silent insertion of statics" could change
                     // initializer values (very bad).
                     if self.qualif.intersects(Qualif::MUTABLE_INTERIOR) {
-                        // Replace MUTABLE_INTERIOR with NOT_CONST to avoid
+                        // A reference of a MUTABLE_INTERIOR place is instead
+                        // NOT_CONST (see `if forbidden_mut` below), to avoid
                         // duplicate errors (from reborrowing, for example).
                         self.qualif = self.qualif - Qualif::MUTABLE_INTERIOR;
-                        self.add(Qualif::NOT_CONST);
                         if self.mode != Mode::Fn {
                             span_err!(self.tcx.sess, self.span, E0492,
                                       "cannot borrow a constant which may contain \
                                        interior mutability, create a static instead");
                         }
+                    } else {
+                        forbidden_mut = false;
```

Comment:
Can you put a comment here? What does this `else` indicate? I can't see it in the diff, it'll also be hard to tell when reading. Something like:

```
// Not a mut borrow
```

(Is that right? I'm guessing)

Reply:
Hmm maybe I should split this part into its own commit, it's a straight-forward translation of the old code, but it's mixed in with new parts.